### PR TITLE
[MIG] event_sale_registration_multi_qty: Migration to 14.0

### PR DESCRIPTION
**Pull request by chienandalu**
In this migration:

- Adapt to new methods and behaviors of the sale order in relation to the registrations.
- Fix the registrations count to sum multi qty registrations.

- [x] Pending on #265 

cc @Tecnativa TT34349

please review @ernestotejeda @pedrobaeza

### CODE REVIEW

File: event_sale_registration_multi_qty/tests/test_event_sale_registration_multi_qty.py [hunk 1]
```diff
@@ -36,72 +35,44 @@ def setUpClass(cls):
...
+            line.product_id = self.product
+            line.event_id = event
+            line.product_uom_qty = qty
+            line.event_ticket_id = event.event_ticket_ids[:1]
+        sale_form.save()
+
+    def _create_sale(self):
+        sale_form = Form(self.env["sale.order"])
+        sale_form.partner_id = self.partner
+        return sale_form.save()
```
Comment (pedrobaeza) [inline event_sale_registration_multi_qty/tests/test_event_sale_registration_multi_qty.py]: For being a bit faster, return the form and don't save until the end.

File: event_sale_registration_multi_qty/wizards/event_edit_registration.xml [hunk 2]
```diff
@@ -7,7 +7,7 @@
         <field name="inherit_id" ref="event_sale.view_event_registration_editor_form" />
         <field name="arch" type="xml">
             <field name="name" position="before">
-                <field name="qty" />
+                <field name="qty" readonly="1" />
```
Comment (pedrobaeza) [inline event_sale_registration_multi_qty/wizards/event_edit_registration.xml:10]: force_save=1?

**Review by pedrobaeza — APPROVED** [review]

**Review by rafaelbn — APPROVED** [review]

**Pull request merged**